src/layout/RepeatingGroup/RepeatingGroupContext.tsx

```diff
@@ -124,6 +128,87 @@ function produceStateFromNode(node: BaseLayoutNode<CompRepeatingGroupInternal>):
   };
 }
 
+type PaginationState =
+  | {
+      hasPagination: true;
+      currentPage: number;
+      totalPages: number;
+      rowsPerPage: number;
+      rowsToDisplay: Row[];
+    }
+  | {
+      hasPagination: false;
+      currentPage: undefined;
+      totalPages: undefined;
+      rowsPerPage: undefined;
+      rowsToDisplay: Row[];
+    };
+
+/**
+ * Produces the current pagination state if relevant
+ */
+function producePaginationState(
+  currentPage: number | undefined,
+  node: BaseLayoutNode<CompRepeatingGroupInternal>,
```

Comment:
Not really used that much, but in principle the component can override `BaseLayoutNode` and provide its own implementation. I do that in the `hierarchy-generator-rewrite` branch, but will probably revert that particular change down the road (as I don't seem to need it anyway), but I did have to go through lots of places and correct the code to point to LayoutNode instead of BaseLayoutNode.

```suggestion
  node: LayoutNode<'RepeatingGroup'>,
```

Reply:
Yeah I pretty much copied from:
```typescript
function produceStateFromNode(node: BaseLayoutNode<CompRepeatingGroupInternal>): NodeState {
```